sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -604,6 +604,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     }.toList
   }
 
+  // TODO: Fix toJSON so that we can more safely handle Map and Seq with loop.
```

Comment:
I think it is better to make the comment self-contained. So, readers of this part do not need to guess or search the jira to understand what this line means.